rotkehlchen/globaldb/updates.py

```diff
@@ -159,6 +159,8 @@ def __init__(self, msg_aggregator: 'MessagesAggregator') -> None:
         self.common_asset_details_re = re.compile(r'.*INSERT +INTO +common_asset_details\( *identifier *, *symbol *, *coingecko *, *cryptocompare *, *forked *, *started *, *swapped_for *\) *VALUES\((.*?),(.*?),(.*?),(.*?),(.*?),([^,]*?),([^,]*?)\).*')  # noqa: E501
         self.assets_collection_re = re.compile(r'.*INSERT +INTO +asset_collections\( *id *, *name *, *symbol *\) *VALUES +\(([^,]*?),([^,]*?),([^,]*?)\).*?')  # noqa: E501
         self.multiasset_mappings_re = re.compile(r'.*INSERT +INTO +multiasset_mappings\( *collection_id *, *asset *\) *VALUES +\(([^,]*?), *([\'"])([^,]+?)\2\).*?')  # noqa: E501
+        if self.local_assets_version > 31:  # multiasset_mappings table adds is_main_asset column to identify main token in collections  # noqa: E501
```

Comment:
> Shouldn't this be an if/else? with the code above?

oh yeah, it should. fixed that.

> Also do we do this? Don't we just keep the version of the code in sync with the latest asset update and then just make sure the asset update does not work for versions less than a specific global DB version?

there's a need for both regex patterns.

```py
re.compile(r'.*INSERT +INTO +multiasset_mappings\( *collection_id *, *asset *\) *VALUES +\(([^,]*?), *([\'"])([^,]+?)\2\).*?')  # noqa: E501
```
is needed to add assets from version 16 -> 30 and the new regex pattern cannot be used as the default since it'll break old updates.